FS19/ML/Zusammenfassung_ML.tex

```diff
@@ -1488,4 +1491,193 @@ \subsubsection{k-fold Cross Validation}
 	\item Pick the model $M-K$ with the lowest estimated generalisation error and retrain that model on the entirety of the training set $S$.
 \end{enumerate}
 
+\section{Clustering and Association Rules}
+
+\subsection{k-Means Algorithm}
+
+The k-Means algorithm is an algorithm that automatically identifies cluster centers. One has to specify the amount of clusters one wants before it can start though. The algorithm functions as follows:
+
+\begin{enumerate}
+	\item Input: Number of clusters $k>0$ and data points $x_1,...,x_n \in \mathbb{R}^m$
+	\item Randomly choose $k$ cluster centers $\mu_1,...,\mu_k \in \mathbb{R}^m$
+	\item Repeat until convergence
+	\begin{enumerate}
+		\item Assign each data point $x_i$ to its neares cluster center $\mu_j$
+		\begin{equation*}
+			c_i = \underset{j\in\{1,...,k\}}{\text{argmin}}\ \| x_i - \mu_j \|^2
+		\end{equation*}
+		\item  Update each cluster center to the mean of all assigned data points
+		\begin{equation*}
+			\mu_j := \frac{\sum_{i:c_i=j} x_i}{|\{i:c_i = j\}|}
+		\end{equation*}
+	\end{enumerate}
+\end{enumerate}
+
+\subsubsection{Addendum}
+\begin{align*}
+	x &= (x_1,x_2)
+	\| x \| &= \sqrt{x_1^2 + x_2^2}
+	\| x \|^2 &= x_1^2 + x_2^2
+	y = (y_1, y_2)
+	\| x - y \| &= \sqrt{(x_1 - y_1)^2 + (x_2 - y_2)^2}
+\end{align*}
+
+\subsubsection{Clustering Distortion}
+
+The total distortion can be measured by summing up the squared distance between each point and it's cluster center
+\begin{equation}
+	\sum_{i=1}^{n}\| x_i -\mu_{c_i} \|^2
+\end{equation}
+
+The average distortion per data point is
+\begin{equation}
+	\frac{1}{n}\sum_{i=1}^{n}\| x_i -\mu_{c_i} \|^2
+\end{equation}
+
+Average distortion allows to compare clusterings over different data sets
+
+\subsubsection{Convergence and Optimality}
+
+Optimal clustering minimises the total distortion, which is a NP-hard problem for $x>1$. Therefore k-Means only approximates the optimal solution, but always converges, albeit not necessarily to a global minimum. In practice k-Means is executed several time and the clustering with minimum distortion is chosen.
+
+\subsubsection{Choose the Number of Clusters}
+
+The choice of clusters balances between low distortion and fewest number of clusters. The elbow method finds a good compromise in general (see figure \ref{fig:elbowmethod}).
+
+\begin{figure}[htb!]
+	\centering
+	\includegraphics[keepaspectratio, width=0.4\linewidth]{Pictures/elbow_method}
+	\caption{The elbow method to find the best number of clusters}
+	\label{fig:elbowmethod}
+\end{figure}
+
+\section{Association}
+
+An \textbf{association rule} is an implication $X\rightarrow Y$, where $X$ and $Y$ are disjoint item sets (refer to \ref{fig:associationrule}).
+
+\begin{figure}[htb!]
+	\centering
+	\includegraphics[keepaspectratio, width=0.7\linewidth]{Pictures/association_rule}
+	\caption{A graphical representation of an Association Rule Learner}
+	\label{fig:associationrule}
+\end{figure}
+
+\subsubsection{Support of a Set of Items}
```

Comment:
Why subsubsection instead of just subsection?

Reply:
Because I'm stupid